ArrayList/LinkedList.cs

```diff
@@ -4,7 +4,656 @@
+                }
+            }
+            else
+            {
+                _root = null;
+                _tail = _root;
+            }
+        }
+
+        public void Revers()
+        {
+            if (Length > 0)
+            {
+                Node<T> previous = null;
+                Node<T> current = _root;
+                Node<T> following = _root;
+
+                while (!(current is null!))
+                {
+                    following = following.Next;
+                    current.Next = previous;
+                    previous = current;
+                    current = following;
+                }
+
+                current = _root;
+                _root = _tail;
+                _tail = current;
+            }
+        }
+
+        public int GetIndex(T value)
+        {
+            if (value != null)
+            {
+                if (Length > 0)
+                {
+                    Node<T> current = _root;
+                    int index = 0;
+
+                    while (!(current is null))
+                    {
+                        if (current.Value.CompareTo(value) == 0)
+                        {
+                            return index;
+                        }
+
+                        ++index;
+                        current = current.Next;
+                    }
+
+                    return -1;
+                }
+
+                return -1;
+            }
+            else
+            {
+                throw new ArgumentException("Invalid value");
+            }
+        }
+
+        public int GetIndexMinValue()
+        {
+            if (Length > 0)
+            {
+                Node<T> current = _root;
+                T minValue = current.Value;
+                int indexMinValue = 0;
+
+                for (int i = 0; i < Length; i++)
+                {
+                    if (current.Value.CompareTo(minValue) == -1)
+                    {
+                        minValue = current.Value;
+                        indexMinValue = i;
+                    }
+
+                    current = current.Next;
+                }
+
+                return indexMinValue;
+            }
+
+            return -1;
+        }
+        public int GetIndexMaxValue()
+        {
+            if (Length > 0)
+            {
+                Node<T> current = _root;
+                T maxValue = current.Value;
+                int indexMaxValue = 0;
+
+                for (int i = 0; i < Length; i++)
+                {
+                    if (current.Value.CompareTo(maxValue) == 1)
+                    {
+                        maxValue = current.Value;
+                        indexMaxValue = i;
+                    }
+
+                    current = current.Next;
+                }
+
+                return indexMaxValue;
+            }
+
+            return -1;
+        }
+
+        public T GetMin()
+        {
+            return this[GetIndexMinValue()];
```

Comment:
Unfortunately for linked list it's bad approach because of two loops through collection

Reply:
corrected